### PR TITLE
Bump Python versions used in wheel build

### LINKED ISSUE #159: Bump Python versions
Currently, we are using old Python versions e.g. 3.6.0 and 3.7.0 and hitting bugs several times.

For example following bug happens with Python 3.6.0 + pip 21.2.2.:

```
00:29:26.442460 STDOUT 2392]	[VerifierAgent] [Wed Aug  4 02:32:34 2021]: Using Python 3.6.0	
00:29:26.442473 STDOUT 2392]	[VerifierAgent] [Wed Aug  4 02:32:34 2021]: Installing distribution...	
00:29:26.442483 STDOUT 2392]	[VerifierAgent] [Wed Aug  4 02:32:34 2021]: Running command: ('pyenv', 'exec', 'python', '-m', 'pip', 'install', '-v', '--user', 'cupy_cuda100-10.0.0a2-cp36-cp36m-manylinux1_x86_64.whl')	
00:29:26.903116 STDOUT 2392]	Using pip 21.2.2 from /opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip (python 3.6)	
00:29:26.959541 STDOUT 2392]	Processing ./cupy_cuda100-10.0.0a2-cp36-cp36m-manylinux1_x86_64.whl	
00:29:26.973712 STDERR 2392]	ERROR: Exception:	
00:29:26.973727 STDERR 2392]	Traceback (most recent call last):	
00:29:26.973741 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 173, in _main	
00:29:26.973897 STDERR 2392]	    status = self.run(options, args)	
00:29:26.973900 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/cli/req_command.py", line 203, in wrapper	
00:29:26.973904 STDERR 2392]	    return func(self, options, args)	
00:29:26.973939 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 316, in run	
00:29:26.973941 STDERR 2392]	    reqs, check_supported_wheels=not options.target_dir	
00:29:26.973946 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 95, in resolve	
00:29:26.974025 STDERR 2392]	    collected.requirements, max_rounds=try_to_avoid_resolution_too_deep	
00:29:26.974028 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 472, in resolve	
00:29:26.974050 STDERR 2392]	    state = resolution.resolve(requirements, max_rounds=max_rounds)	
00:29:26.974051 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 341, in resolve	
00:29:26.974054 STDERR 2392]	    self._add_to_criteria(self.state.criteria, r, parent=None)	
00:29:26.974079 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 168, in _add_to_criteria	
00:29:26.974084 STDERR 2392]	    candidates=build_iter_view(matches),	
00:29:26.974139 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_vendor/resolvelib/structs.py", line 164, in build_iter_view	
00:29:26.974141 STDERR 2392]	    matches = list(matches)	
00:29:26.974144 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 417, in <genexpr>	
00:29:26.974167 STDERR 2392]	    and all(req.is_satisfied_by(c) for req in requirements[identifier])	
00:29:26.974170 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 417, in <genexpr>	
00:29:26.974191 STDERR 2392]	    and all(req.is_satisfied_by(c) for req in requirements[identifier])	
00:29:26.974193 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/requirements.py", line 39, in is_satisfied_by	
00:29:26.974245 STDERR 2392]	    return candidate == self.candidate	
00:29:26.974250 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 172, in __eq__	
00:29:26.974301 STDERR 2392]	    return links_equivalent(self._link, other._link)	
00:29:26.974310 STDERR 2392]	  File "/opt/pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/_internal/models/link.py", line 288, in links_equivalent	
00:29:26.974404 STDERR 2392]	    return _CleanResult.from_link(link1) == _CleanResult.from_link(link2)	
00:29:26.974406 STDERR 2392]	AttributeError: type object '_CleanResult' has no attribute 'from_link'	
```

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #159.

### CODE REVIEW

File: dist_config.py [hunk 1]
```diff
@@ -304,38 +304,28 @@
 # Key-value of python version (used in pyenv) to use for build and its
```
Comment (kmaehashi) [inline dist_config.py:304]:
```suggestion
# Key-value of python version to use for build and its
```

**Comment by emcastillo** [comment]
```
pyenv: version `3.7' is not installed (set by PYENV_VERSION environment variable)
```
:D

**Review by emcastillo — APPROVED** [review]

**Pull request merged**